File: Server/BookMyVenue.Core/Entities/Venue.cs
using BookMyVenue.Core.Enums;

namespace BookMyVenue.Core.Entities;

public class Venue : BaseEntity
{
    public Guid OwnerId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string Type { get; set; } = string.Empty;
    public string Address { get; set; } = string.Empty;
    public string City { get; set; } = string.Empty;
    public decimal PricePerDay { get; set; }
    public int Capacity { get; set; }
    public VenueStatus Status { get; set; } = VenueStatus.Pending;
    public DateOnly AvailableFrom { get; set; }
    public DateOnly AvailableTo { get; set; }

    // Payment policy
    public bool IsAdvanceRequired { get; set; } = false;
    public decimal? AdvancePercentage { get; set; }
    public int? BalanceDueDaysBeforeEvent { get; set; }

    // Cancellation policy
    public bool IsCancellationAllowed { get; set; } = false;
    public int? CancellationDeadlineDays { get; set; }

    // Navigation properties
    public VenueOwnerProfile Owner { get; set; } = null!;
    public ICollection<VenueImage> Images { get; set; } = new List<VenueImage>();
    public ICollection<VenueAmenity> Amenities { get; set; } = new List<VenueAmenity>();
    public ICollection<VenueBlockedDate> BlockedDates { get; set; } = new List<VenueBlockedDate>();
    public ICollection<Booking> Bookings { get; set; } = new List<Booking>();
}

File: client/types/venue.ts
export interface BookedDateRange {
  startDate: string;
  endDate: string;
}

export interface BlockedDateRange {
  startDate: string;
  endDate: string;
  blockedReason: string;
}

export interface Venue {
  id: string;
  name: string;
  description: string;
  type: string;
  address: string;
  city: string;
  pricePerDay: number;
  capacity: number;
  availableFrom: string;
  availableTo: string;
  bookedDates: BookedDateRange[];
  blockedDates: BlockedDateRange[];
  status: string;
  amenities: string[];
  images: string[];

  isAdvanceRequired: boolean;
  advancePercentage: number | null;
  balanceDueDaysBeforeEvent: number | null;

  isCancellationAllowed: boolean;
  cancellationDeadlineDays: number | null;
}

export interface CreateVenueRequest {
  name: string;
  description: string;
  type: string;
  address: string;
  city: string;
  pricePerDay: number;
  capacity: number;
  availableFrom: string;
  availableTo: string;
  amenities: string[];
  images?: string[];
  isAdvanceRequired: boolean;
  advancePercentage?: number;
  balanceDueDaysBeforeEvent?: number;
  isCancellationAllowed: boolean;
  cancellationDeadlineDays?: number;
}

export interface VenueSearchParams {
  city?: string;
  type?: string;
  capacity?: number;
  maxPrice?: number;
  startDate?: string;
  endDate?: string;
}